dev/com.ibm.ws.opentracing.cdi/src/com/ibm/ws/opentracing/cdi/TracedInterceptor.java

```diff
@@ -0,0 +1,169 @@
+/*******************************************************************************
+ * Copyright (c) 2017 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.opentracing.cdi;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Method;
+
+import javax.annotation.Priority;
+import javax.interceptor.AroundInvoke;
+import javax.interceptor.Interceptor;
+import javax.interceptor.InvocationContext;
+import javax.ws.rs.HttpMethod;
+import javax.ws.rs.Path;
+
+import org.eclipse.microprofile.opentracing.Traced;
+
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+import com.ibm.ws.opentracing.OpentracingService;
+import com.ibm.ws.opentracing.OpentracingTracerManager;
+
+import io.opentracing.ActiveSpan;
+import io.opentracing.Span;
+import io.opentracing.Tracer;
+
+/**
+ * {@code #executeFT(InvocationContext)} is called for every method with the {@code Traced} annotation
+ * or every method of a class with the {@code Traced} annotation. This interceptor is run after platform
+ * interceptors but before application interceptors using {@code Interceptor.Priority.LIBRARY_BEFORE}.
+ * If we figure out that the method in question is a JAXRS method, then we skip it because we know
+ * it has been processed by {@code OpentracingContainerFilter} (which has the requisite knowledge
+ * about the incoming URI which this class won't have).
+ */
+@Traced
+@Interceptor
+@Priority(Interceptor.Priority.LIBRARY_BEFORE)
+public class TracedInterceptor {
+
+    private static final TraceComponent tc = Tr.register(TracedInterceptor.class);
+
+    /**
+     * See class comment.
+     *
+     * @param context Information about the wrapped method.
+     * @return Result of the invoking the method.
+     * @throws Exception Thrown by wrapped method.
+     */
+    @AroundInvoke
+    public Object executeFT(InvocationContext context) throws Exception {
```

Comment:
executeFT should be just execute. I think this is copy and paste error from Fault Tolerance:o.